From 41024f70f940893525a4e47859c5516b0c000035 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Wed, 13 Jun 2018 19:45:58 +0800
Subject: [PATCH 4257/5242] MLK-18605-4 drm/bridge: sec-dsim: refine data
 lanes stop state check

commit  beeeb71dd242a3f02051b9867f0abcb998265a51 from
https://source.codeaurora.org/external/imx/linux-imx.git

When the attached dsi device does not use all the data lanes
to transfer data, data lanes stop state check should only
check the lanes used precisely, since unused lanes state is
not guaranteed to be in some stable state.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 6787ee8505ab16bf7bba38c721da0bfa87e9de0e)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/bridge/sec-dsim.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/bridge/sec-dsim.c b/drivers/gpu/drm/bridge/sec-dsim.c
index f44d6c8..6752ccc 100644
--- a/drivers/gpu/drm/bridge/sec-dsim.c
+++ b/drivers/gpu/drm/bridge/sec-dsim.c
@@ -426,7 +426,7 @@ static int sec_mipi_dsim_bridge_attach(struct drm_bridge *bridge)
 static int sec_mipi_dsim_config_pll(struct sec_mipi_dsim *dsim)
 {
 	int ret;
-	uint32_t pllctrl = 0, status, data_lanes_en;
+	uint32_t pllctrl = 0, status, data_lanes_en, stop;
 
 	dsim_write(dsim, 0x8000, DSIM_PLLTMR);
 
@@ -451,7 +451,8 @@ static int sec_mipi_dsim_config_pll(struct sec_mipi_dsim *dsim)
 		return -EBUSY;
 	}
 
-	if (STATUS_GET_STOPSTATEDAT(status) != data_lanes_en) {
+	stop = STATUS_GET_STOPSTATEDAT(status);
+	if ((stop & data_lanes_en) != data_lanes_en) {
 		dev_err(dsim->dev,
 			"one or more data lanes is not in stop state\n");
 		return -EBUSY;
-- 
1.7.9.5

